### ISSUE TITLE
[self-hosted] image build fails: cannot run context initializer (buildBaseImage) – cannot initialize workspace: git initializer: git clone [...] failed

### ISSUE THREAD

**Issue by rsliotta** [gitpod-io/gitpod#3016]
### Describe the bug
Trying to checkout a project with a custom dockerfile based on workspace full

### Steps to reproduce

Just try to create the workspace

### Expected behavior

Get a workspace

### Additional information

If I remove the image tag in the .gitpod.yaml  it works.  I have been using this on the commerical gitpod site.  It's my testing with the self-hosted that does not work.  I have tried a number of projects.  

### Example repository

Logs:

```
{"@type":"type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent","buildRef":"registry.bobdot.me:4443/gitpod/workspace-images:d4234bb600328ee17375e8f1b6935dd53b18330e887defe7993122a564edcf9f","error":"cannot run context initializer:\n    github.com/gitpod-io/gitpod/image-builder/pkg/builder.(*DockerBuilder).buildBaseImage\n        /tmp/build/components-image-builder--app.b18e4ba7f75dbc19795882a725d62efee8ca3cd2/pkg/builder/builder.go:491\n  - container exited with non-zero exit code: 1:\n    github.com/gitpod-io/gitpod/image-builder/pkg/builder.(*DockerBuilder).runContainer\n        /tmp/build/components-image-builder--app.b18e4ba7f75dbc19795882a725d62efee8ca3cd2/pkg/builder/docker.go:98","level":"warning","message":"build failed","serviceContext":{"service":"image-builder","version":""},"severity":"WARNING","time":"2021-01-26T01:27:18Z"}
{"buildRef":"registry.bobdot.me:4443/gitpod/workspace-images:d4234bb600328ee17375e8f1b6935dd53b18330e887defe7993122a564edcf9f","level":"info","message":"deregistering build","serviceContext":{"service":"image-builder","version":""},"severity":"INFO","status":"done_failure","time":"2021-01-26T01:27:18Z"}
{"level":"info","logs":"building base image\n\ninitializing base image context\n\u0002\u0000\u0000\u0000\u0000\u0000\u0001\\\u001b[31minit failed                                   \u001b[31merror\u001b[0m=cannot initialize workspace: git initializer: git clone https://github.com/BobDotMe/bob.git . failed (exit status 128): Cloning into '.'...\nfatal: unable to access 'https://github.com/BobDotMe/bob.git/': Operation timed out after 300023 milliseconds with 0 out of 0 bytes received\n\n","message":"build was closed but not successful","serviceContext":{"service":"image-builder","version":""},"severity":"INFO","status":3,"time":"2021-01-26T01:27:18Z"}
{"level":"debug","message":"closing build listener","serviceContext":{"service":"image-builder","version":""},"severity":"DEBUG","time":"2021-01-26T01:27:18Z"}
{"level":"debug","message":"closing build listener","serviceContext":{"service":"image-builder","version":""},"severity":"DEBUG","time":"2021-01-26T01:27:18Z"}
```

**Comment by wulfthimm** [comment]
Hi @rsliotta,
please do not use `sudo` in the Dockerfile. Run the commands which need root access as `USER root` and end the file with `USER gitpod` so that the workspace is started with `gitpod` user.

**Comment by rsliotta** [comment]
I can try this but I believe I tested with just the image and it still fails.  I will retest and come back.

**Comment by rsliotta** [comment]
Same result.....

```
{"@type":"type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent","buildRef":"registry.bobdot.me:4443/gitpod/workspace-images:ba3c2e2adce8ef873afd957438620fb5e486ed68bae5d0e71a6541f08b432410","error":"cannot run context initializer:\n    github.com/gitpod-io/gitpod/image-builder/pkg/builder.(*DockerBuilder).buildBaseImage\n        /tmp/build/components-image-builder--app.b18e4ba7f75dbc19795882a725d62efee8ca3cd2/pkg/builder/builder.go:491\n  - container exited with non-zero exit code: 1:\n    github.com/gitpod-io/gitpod/image-builder/pkg/builder.(*DockerBuilder).runContainer\n        /tmp/build/components-image-builder--app.b18e4ba7f75dbc19795882a725d62efee8ca3cd2/pkg/builder/docker.go:98","level":"warning","message":"build failed","serviceContext":{"service":"image-builder","version":""},"severity":"WARNING","time":"2021-01-26T19:52:36Z"}
{"buildRef":"registry.bobdot.me:4443/gitpod/workspace-images:ba3c2e2adce8ef873afd957438620fb5e486ed68bae5d0e71a6541f08b432410","level":"info","message":"deregistering build","serviceContext":{"service":"image-builder","version":""},"severity":"INFO","status":"done_failure","time":"2021-01-26T19:52:36Z"}
{"level":"info","logs":"building base image\n\ninitializing base image context\n\u0002\u0000\u0000\u0000\u0000\u0000\u0001v\u001b[31minit failed                                   \u001b[31merror\u001b[0m=cannot initialize workspace: git initializer: git clone https://github.com/BobDotMe/bob-ubuntu-image.git . failed (exit status 128): Cloning into '.'...\nfatal: unable to access 'https://github.com/BobDotMe/bob-ubuntu-image.git/': Operation timed out after 300017 milliseconds with 0 out of 0 bytes received\n\n","message":"build was closed but not successful","serviceContext":{"service":"image-builder","version":""},"severity":"INFO","status":3,"time":"2021-01-26T19:52:36Z"}
{"level":"debug","message":"closing build listener","serviceContext":{"service":"image-builder","version":""},"severity":"DEBUG","time":"2021-01-26T19:52:36Z"}
```

Now if I remove the image lines from the .gitpod.yaml it works.  So's something about the docker or image as it CAN reach github fine without.

**Comment by rsliotta** [comment]
After doing a fair amount of testing i found:

1. if you mention the gitpod/workspace-full in the .gitpod.yaml and no file: entry, it indeed DOES create a custom image ad starts.  Of course then you cannot customize the container.
2. If you add the image: and file: in the .gitpod.yaml and point it to a dockerfile it simply hangs.  
3. If you remove the image: and file: from the .gitpod.yaml it starts.
4. I tried a Dockerfile with simply a FROM: gitpod/workspace-full and it hangs as well. 
5. Finally I tried an EMPTY Dockerfile and it still hangs. 
6. I believe the issue is with the custom dockerfile handoff.

**Comment by corneliusludmann** [comment]
At first, it's totally fine to use `sudo` in `.gitpod.Dockerfile`.

However, I'm not sure if I get your issue right, @rsliotta. I opened your repo https://gitpod.io/#https://github.com/bobdotme/bob-ubuntu-image (revision [`ea9eea7`](https://github.com/BobDotMe/bob-ubuntu-image/commit/ea9eea7a5fffecb350017463ae70f3cb833c6cb1)) and it worked just fine for me. The image builds and the workspace starts without problems. Could you please tell us more about your issue?

**Comment by corneliusludmann** [comment]
Ah, after having a closer look at your logs, it seems that you're talking about a self-hosted installation, right? @rsliotta 

It seems that the image builder is not able to build your custom workspace image. Is there something in your `image-builder` logs?

**Comment by rsliotta** [comment]
Just the same logs I posted above each time.

It only hangs when I use the .gitpod.dockerfile

If I add the image to the .gitpod.yaml it does build the custom image.

It's very strange....

It's version 0.6.0

**Comment by rsliotta** [comment]
Attached are both the service and dind logs.
[ib-dind.log](https://github.com/gitpod-io/gitpod/files/5880509/ib-dind.log)
[ib-service.log](https://github.com/gitpod-io/gitpod/files/5880510/ib-service.log)

**Comment by rsliotta** [comment]
![gitpod-error](https://user-images.githubusercontent.com/44572134/106069776-216aa280-60d1-11eb-83a3-28019017dd42.png)

**Comment by rsliotta** [comment]
Guys.....  I got it to work now.  
Interesting thing.  

When you just use the image: tag it looks like it just uses the image and starts as I mentioned about.  

I figured out when you use the file: .gitpod.dockerfile, that's when DIND is invoked and that is the difference.  That is when the timeouts happen i mentioned above.

Upon a fair amount of digging, I found I needed to set a value of 1400 for dindMTU in the values.yaml.  Apparently this is kind of common issue with  DIND and this was new to me.  

My guess is many people run jumbo frames on their networks so they might not run into this.  I use MTU of 1500 that makes k8S 1450 hence the 2400 for DIND

You might want to mention the importance of this in the documentation.

Hopefully this can help someone else as I see some similar issues with different subjects
